sklearn/utils/estimator_checks.py

```diff
@@ -2068,7 +2070,10 @@ def check_estimators_pickle(name, estimator_orig, readonly_memmap=False):
     estimator.fit(X, y)
 
     if readonly_memmap:
-        unpickled_estimator = create_memmap_backed_data(estimator)
+        with TemporaryDirectory(prefix="sklearn_test_pickle_") as tmpdir:
+            pickle_filepath = Path(tmpdir) / "estimator.pkl"
+            joblib.dump(estimator, pickle_filepath)
+            unpickled_estimator = joblib.load(pickle_filepath, mmap_mode="r")
```

Comment:
Since this is a permission denied, is there still a reference to an object still leaving here?

Reply:
Yes I am working on a fix. Note that there are other things to fix before merging this PR as I said in the description. I will put this PR in draft mode while working on it.